docs/source/about_dataset_load.rst

```diff
@@ -0,0 +1,92 @@
+
+   It returns a list of :class:`datasets.SplitGenerator`. The :class:`datasets.SplitGenerator` contains the name of the split, and keyword arguments that are provided to the :func:`datasets.DatasetBuilder._generate_examples` method. The keyword arguments can be specific to each split, and typically comprise at least the local path to the data files to load for each split.
+
+   .. tip::
+
+       :func:`datasets.DownloadManager.download_and_extract` can download files from a wide range of sources. If the data files are hosted on a special access server, you should use :func:`datasets.DownloadManger.download_custom`. Refer to the package reference of :class:`datasets.DownloadManager` for more details.
+
+3. :func:`datasets.DatasetBuilder._generate_examples` reads and parses the data files for a split, and yields examples with the format specified in the ``features`` from :func:`datasets.DatasetBuilder._info`. The input of :func:`datasets.DatasetBuilder._generate_examples` is the ``filepath`` provided in the last method. 
+
+   The dataset is generated with a Python generator, which doesn't load all the data in memory. As a result, the generator can handle large datasets. However, before the generated samples are flushed to the dataset file on disk, they are stored in an ``ArrowWriter`` buffer. This means the generated samples are written by batch. If your dataset samples consumes a lot of memory (images or videos), then make sure to specify a low value for the ``DEFAULT_WRITER_BATCH_SIZE`` attribute in :class:`datasets.DatasetBuilder`. We recommend not exceeding a size of 200MB.
+
+
+Maintaining integrity
+---------------------
+
+To ensure a dataset is complete, :func:`datasets.load_dataset` will perform some tests on the downloaded files to make sure everything is there. This way, you don't encounter any nasty surprises when your requested dataset doesn't get generated as expected. :func:`datasets.load_dataset` verifies:
+
+* the list of downloaded files
+* the number of bytes of the downloaded files
+* the SHA256 checksums of the downloaded files
+* the number of splits in the generated ``DatasetDict``
+* the number of samples in each split of the generated ``DatasetDict``
+
+TO DO: Explain why you would want to disable the verifications or override the information used to perform the verifications.
```

Comment:
May be helpful to explain why a user would want to disable the verifications or override the info.